fabric/src/main/java/com/discordsrv/fabric/player/FabricPlayer.java

```diff
@@ -67,17 +69,22 @@ public CompletableFuture<Void> kick(Component component) {
 
     @Override
     public void addChatSuggestions(Collection<String> suggestions) {
-        // API not available in Fabric
+        ChatSuggestionsS2CPacket packet = new ChatSuggestionsS2CPacket(ChatSuggestionsS2CPacket.Action.ADD, new ArrayList<>(suggestions));
+        player.networkHandler.sendPacket(packet);
     }
 
     @Override
     public void removeChatSuggestions(Collection<String> suggestions) {
-        // API not available in Fabric
+        ChatSuggestionsS2CPacket packet = new ChatSuggestionsS2CPacket(ChatSuggestionsS2CPacket.Action.REMOVE, new ArrayList<>(suggestions));
+        player.networkHandler.sendPacket(packet);
     }
 
     @Override
     public @Nullable SkinInfo skinInfo() {
-        // Unimplemented
+        MinecraftProfileTextures textures = discordSRV.getServer().getSessionService().getTextures(player.getGameProfile());
+        if (!textures.equals(MinecraftProfileTextures.EMPTY) && textures.skin() != null) {
+            return new SkinInfo(textures.skin().getHash(), null);
```

Comment:
This looks very suspect, and is lacking the skin model

If one of these methods gives the raw base64 texture info, https://github.com/DiscordSRV/Ascension/blob/d348fdc3074940829acf335fabded0e3c16eb609/common/src/main/java/com/discordsrv/common/abstraction/player/provider/model/Textures.java#L54 can be used to parse it